oex_plugin/openedx/core/djangoapps/user_api/accounts/api.py

```diff
@@ -124,4 +125,18 @@ def _update_additional_extended_profile_fields_if_needed(data, user):
     """
     if 'english_proficiency' in data:
         user.extended_profile.english_proficiency = data['english_proficiency']
-        user.extended_profile.save()
+    if 'organization' in data:
+        if org_data := data['organization']:
+            org_label = org_data.get('org_label')
+            try:
+                organization = Organization.objects.get(label__iexact=org_label)
+            except Organization.DoesNotExist:
+                organization = Organization(label=org_label)
+            organization.org_type = org_data.get('org_type')
+            organization.is_organization_registered = org_data.get('is_org_registered')
+            organization.total_employees = org_data.get('total_employees')
+            organization.save()
```

Comment:
can we use get_or_create here?

Reply:
what restricts you from using get_or_create with __iexact?

Reply:
and passing the exact name you want to use for creation to the defaults dict?

Reply:
In my opinion, my solution seems to be the most optimal and understandable for these conditions.

Reply:
How do you like this option:
```
organization, created = Organization.objects.get_or_create(
    label__iexact=org_label,
    defaults={
        'label': org_label,
        'org_type': org_data.get('org_type'),
        'is_organization_registered': org_data.get('is_org_registered'),
        'total_employees': org_data.get('total_employees'),
    }
)

if not created:
    organization.update(
         org_data=org_data.get('org_type')
         is_organization_registered=org_data.get('is_org_registered')
         total_employees=org_data.get('total_employees')
    )
```

or

```
            try:
                organization = Organization.objects.get(label__iexact=org_label)
            except Organization.DoesNotExist:
                organization = Organization.objects.create(
                    label=org_label,
                    org_type=org_data.get('org_type')
                    is_organization_registered=org_data.get('is_org_registered')
                    total_employees=org_data.get('total_employees')
                )
```

Reply:
Also, is there a possibility that we'll receive the same org fields for updates? Maybe we should skip additional DB calls in such a case?

Reply:
my proposal is not valid because update works for query sets only